src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,55 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = findKey(key);
+
+        if (index != -1) {
+            values[index] = value;
+        } else {
+            if (size >= MAX_SIZE) {
+                throw new IllegalStateException("Storage is full");
```

Comment:
You can make single method like this 
```suggestion
private boolean isFull() {
            if (size >= MAX_SIZE) {
                throw new IllegalStateException("Storage is full");
                }
                return false;
                }
                
```
and call this method in "if" block. If your storage is full , you throw exception. But if you return false , your code on 27 - 29 lines will work